elements/rh-health-index/rh-health-index.ts

```diff
@@ -0,0 +1,81 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+
+import styles from './rh-health-index.css';
+
+/**
+ * Health index displays a health grade (A–F) for a particular item or package.
+ *
+ * @summary     Displays a health grade for a particular item or package
+ *
+ */
+@customElement('rh-health-index')
+export class RhHealthIndex extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * Sets the size of the health index
+   * Defaults to `md`
+   */
+  @property({ reflect: true }) size: 'sm' | 'md' | 'lg' | 'xl' = 'md';
+
+  /**
+   * Sets the health grade
+   * Defaults to `A`
+   */
+  @property({
+    reflect: true,
+    attribute: 'grade',
+  })
+    grade: 'A' | 'B' | 'C' | 'D' | 'E' | 'F' = 'A';
+
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  render() {
+    const classes = {
+      [this.grade.toLowerCase()]: !!this.grade,
+    };
+    const boxes = ['A', 'B', 'C', 'D', 'E', 'F'];
+
+    if (this.size === 'sm') {
```

Comment:
this should either be one long if / else chain or one big nested ternary. i'd go tern, personally

```js
const { on, size, active, box } = this;
return html`
  <div id="container"
       class="${classMap({ [size ?? '']: !!size,
                           [on ?? '']: !!on,
                           active, 
                           box })>${this.size === 'sm' ? html`
    ...
  ` : this.size === 'md' ? html`
    ...
  ` : html`
    ...
  `}
  </div>
`;
```

That would also make it easier to use shadow classes `active`, `box`, `sm`, and `md`, etc instead of `:host([size="sm"])` selectors

Reply:
Fixed in next commit

Reply:
It was nice pairing with you on this